### ISSUE TITLE
option 'avoid flatterning branches' is broken for JSONPath

### ISSUE THREAD

**Issue by chris-kiick-sp** [jenkinsci/generic-webhook-trigger-plugin#268]
Selecting the option to not flatten expression values causes the expression to not be evaluated against the post content at all.

 * Plugin version 1.86.3
 * Jenkins version 2.387.1
 * Your configuration
```
            GenericTrigger(
              causeString: 'PR# $pr_number on $pr_repo from $config_branch to $dest_branch',
              printContributedVariables: true,
              shouldNotFlattern: true,
              printPostContent: true,
              genericHeaderVariables: [
                [key: 'x-github-event', regexpFilter: '']
              ],
              genericVariables: [
                [key: 'action', defaultValue: 'nonpr', value: '$.action'],
                [key: 'pr_repo', value: '$.repository.full_name'],
                [key: 'config_branch', value: '$.pull_request.head.ref'],
                [key: 'dest_branch', value: '$.pull_request.base.ref'],
                [key: 'pr_number', value: '$.number']
              ],
              regexpFilterExpression: "pull_request (opened|reopened|synchronize) XXX YYY",
              regexpFilterText: '$x_github_event $action $pr_repo $dest_branch',
              token: 'ZZZ',
              tokenCredentialId: ''
  }
```

 I don't have a full, small test case, but should be easy to reproduce.

On any GenericVariable using JSONPath expressions in the post content, turn on shouldNotFlattern. The variable will not be expanded. If a default value is set it always uses that, otherwise the variable is left unexpanded in the text.

If you look at resolveJsonPath() method in PostContentParameterResolver.java and follow the logic for when shouldNotFlattern is set, it will always return an empty hash. The 'resolved' string is computed but never returned.

Note all the test cases for JSONPath have shoudNotFlattern set to false. 

BTW, is "flatten" deliberately misspelled as "flattern"?

**Comment by tomasbjerre** [comment]
I took a quick look, and yes it looks like the resolved value is never saved here:
https://github.com/jenkinsci/generic-webhook-trigger-plugin/blob/add7cca58723548ddcb76caebda96eb54978aa8c/src/main/java/org/jenkinsci/plugins/gwt/resolvers/PostContentParameterResolver.java#L94

Im putting this on my todo, will probably have time to fix in this weekend.

**Comment by tomasbjerre** [comment]
I released `1.86.4` that should fix this issue. Both spelling and the falttening.

Thanks for reporting! Open issue again if not working.

**Comment by xabolcs** [comment]
> Both spelling and the **falttening**.

:wink:
